### ISSUE TITLE
error constructing XcodeVersionInfo despite Xcode being installed

### ISSUE THREAD

**Issue by BrianHicks** [facebook/buck2#209]
On the latest binary (`buck2 e0cf22f7f5e736458d848c6bf0f60d67 <local>` in `buck2 --version`) I get the following error:

```
Constructing `XcodeVersionInfo` using host platform MacOS.

Caused by:
    0: deserializing Xcode `version.plist`
    1: Io(Os { code: 2, kind: NotFound, message: "No such file or directory" })
    2: No such file or directory (os error 2)
Build ID: 3cda14ea-393b-4e82-9f45-2d667a1aa670
Jobs completed: 0. Time elapsed: 0.0s.
BUILD FAILED
```

This is on a new repo just after running `buck2 init --git`.

It looks like this might be related to https://github.com/facebook/buck2/issues/191. I have Xcode installed, but it doesn't look like the target path exists.

I'm still on macOS Monterey (12.6.5), which may be relevant here.

```console
$ xcode-select --print-path
/Library/Developer/CommandLineTools
```

Here are places within that path that `version.plist` exists:

```
/Library/Developer/CommandLineTools/usr/lib/sourcekitd.framework/Versions/A/Resources/version.plist
/Library/Developer/CommandLineTools/usr/lib/sourcekitd.framework/Versions/A/XPCServices/SourceKitService.xpc/Contents/version.plist
/Library/Developer/CommandLineTools/usr/lib/sourcekitdInProc.framework/Versions/A/Resources/version.plist
/Library/Developer/CommandLineTools/usr/lib/swift/pm/llbuild/llbuild.framework/Versions/A/Resources/version.plist
/Library/Developer/CommandLineTools/Library/PrivateFrameworks/LLDB.framework/Versions/A/Resources/version.plist
/Library/Developer/CommandLineTools/Library/PrivateFrameworks/LLDB.framework/Versions/A/XPCServices/RootDebuggingXPCService.xpc/Contents/version.plist
/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/Resources/Python.app/Contents/version.plist
/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/Resources/version.plist
```

**Comment by BrianHicks** [comment]
I got around this for now by grabbing the binaries from the build before the fix for #191, so if it didn't start there it's at least sometime after https://github.com/facebook/buck2/actions/runs/4768991258

**Comment by krallin** [comment]
It's probably rather coming from:

https://github.com/facebook/buck2/commit/5accfdc45db1ef3ab95421b7c048699f2fb259c0

Perhaps this whole XCode version thing isn't something we should have in the OSS build. This is a bit of a hack internally already, and it seems like this is doing a lot more harm thangood.
